File: gdx2d-demoDesktop/src/main/java/ch/hevs/gdx2d/demos/shaders/simple/DemoShaderParticles.java
package ch.hevs.gdx2d.demos.shaders.simple

import ch.hevs.gdx2d.desktop.PortableApplication
import ch.hevs.gdx2d.lib.GdxGraphics
import com.badlogic.gdx.Gdx


/**
 * A nice particle shader. Shader code taken from
 * http://glsl.heroku.com/e#13789.0
 *
 * @author Pierre-André Mudry (mui)
 * @version 0.2
 */
class DemoShaderParticles : PortableApplication() {

    private var time = 0f

    override fun onInit() {
        this.setTitle("A nice particle shader demo, mui 2013")
    }

    override fun onGraphicRender(g: GdxGraphics) {
        if (g.shaderRenderer == null)
            g.setShader("shader/particles.fp")

        g.clear()

        // Draws the shader
        time += Gdx.graphics.deltaTime
        g.drawShader(time)

        g.drawStringCentered(windowHeight * 0.95f,
                "Original shader code from http://glsl.heroku.com/e#13789.0")
        g.drawFPS()
        g.drawSchoolLogo()
    }

    companion object {

        @JvmStatic
        fun main(args: Array<String>) {
            DemoShaderParticles()
        }
    }
}

File: gdx2d-demoDesktop/src/main/scala/ch/hevs/gdx2d/demos/shaders/simple/DemoShaderParticles.scala
package ch.hevs.gdx2d.demos.shaders.simple

import ch.hevs.gdx2d.desktop.DesktopApplication
import ch.hevs.gdx2d.lib.GdxGraphics
import com.badlogic.gdx.Gdx

/**
 * A nice particle shader.
 *
 * @author Pierre-André Mudry (mui)
 * @version 2.0
 */
class DemoShaderParticles extends DesktopApplication {

  private var time = 0f

  override def onInit(): Unit = {
    setTitle("A nice particle shader demo, mui 2013")
  }

  override def onGraphicRender(g: GdxGraphics): Unit = {
    if (g.getShaderRenderer == null) {
      g.setShader("shader/particles.fp")
    }

    g.clear()

    time += Gdx.graphics.getDeltaTime
    g.drawShader(time)

    g.drawStringCentered(getWindowHeight * 0.95f, "Original shader code from http://glsl.heroku.com/e#13789.0")
    g.drawFPS()
    g.drawSchoolLogo()
  }
}

object DemoShaderParticles {
  def main(args: Array[String]): Unit = {
    new DemoShaderParticles().launch()
  }
}
